### DISCUSSION TITLE
Creating a meta-samplers repository

### DISCUSSION THREAD

**Discussion by AdrienCorenflos** [blackjax-devs/blackjax#661] [General]
Hi,

I've been thinking recently about splitting responsibilities between sampling algorithms and meta-algorithms. Chief of these being SMC. It's pretty clear that they cannot be used by themselves, with no inner kernel feeding them.

It has some benefits:

- decluttering the API
- signaling responsibilities more clearly
- making contributions more localised 
- being able to create more specialised ecosystems while maintaining coherence

And some inconvenients

- Split download and import (although that can be solved relatively easily via deps)
- Enforcing APIs compatibility would not be immediate 
- Creating upstream monitoring strain for maintenance 

What do you think? 
Adrien

**Comment by junpenglao** [comment]
First thing comes to mind is: Is our SMC API slim enough that user could potentially use it without understanding too deep about the inner kernel implementation?

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

Related update: **tuningfork** (https://github.com/blackjax-devs/tuningfork) is now public and is partly taking on this role — a separate, BlackJAX-native space for benchmarking and composing samplers/meta-algorithms without bloating the core API. Still WIP but coming together; a good place to continue the meta-sampler discussion.
